### PR TITLE
[CI] Update to ubuntu-latest, and fix paths to LLVM.

### PR DESCRIPTION
**Pull request by mikeurbach**
Previously, the broken path to llvm-lit caused builds to fail for
ubuntu-latest, because we were falling back to the binary that came
installed on the OS. By fixing the path, we can update to the latest.

### CODE REVIEW

**Comment by mikeurbach** [comment]
We discussed potentially just caching the `llvm/install` directory, but I'm not sure if that works out of the box. The issue is we use the `llvm-lit` that gets built in `llvm/build`, and even though we use LLVM_INSTALL_UTILS=ON, that doesn't appear to be installed in the `llvm/install` directory at the moment.

For now, I'm not touching this, and continuing to cache the whole `llvm` directory. This is just to fix the paths and update back to `ubuntu-latest`.

File: .github/workflows/assign-reviewers.yml
Comment (teqdruid) [inline .github/workflows/assign-reviewers.yml:11]: FYI: this workflow broke a while back and it wasn't useful enough for me to fix so I just disabled it. It should just be deleted and CODEOWNERs should be put in a readable format.

Comment (teqdruid) [reply .github/workflows/assign-reviewers.yml:11]: Not requesting any changes.

**Review by teqdruid — APPROVED** [review]
Thanks!

File: .github/workflows/assign-reviewers.yml
Comment (mikeurbach) [reply .github/workflows/assign-reviewers.yml:11]: Ah fair enough, just did a find-and-replace here, not worrying about it.

**Comment by mikeurbach** [comment]
Thanks for taking a look @teqdruid, but the workflow still isn't finding llvm-lit just yet. I'll make sure it works before merging. The changes to MLIR_DIR and LLVM_DIR should be good to go, I tested this last night with CMAKE_FIND_DEBUG_MODE.

**Comment by teqdruid** [comment]
Yeah, I noticed the build fail after I approved. I'm good with any changes related to finding llvm-lit (so long as it works).

**Comment by mikeurbach** [comment]
Obviated by https://github.com/llvm/circt/pull/1061

**Pull request closed without merging**
